code/clue.js

```diff
@@ -1,59 +1,171 @@
+  'Living Room', 
+  'Observatory', 
+  'Theatre', 
+  'Guest House', 
+  'Patio'
+]
 
 // THIS FUNCTION WILL RANDOMLY SELECT ONE ITEM FROM THE ARRAY THAT YOU PASS IN TO THE FUNCTION.
-// YOU DON'T NEED TO CHANGE THIS, JUST TRY TO UNDERSTAND IT. AND HOW TO USE IT.
 const randomSelector = array => {
   return array[Math.floor(Math.random() * array.length)]
 }
 
-// CREATE AN OBJECT THAT KEEPS THE MYSTERY.
-// With a killer, a weapon and a room.
-// The values will be set later.
 
-// This function will be invoked when you click on the killer card.
+//OBJECT THAT KEEPS THE MYSTERY.
+const mystery = {
+killer: '',
+weapon: '',
+room: ''
+}
+
+//FUNCTIONS FOR THE PICK CARD KILLER, WEAPON & ROOM.
 const pickKiller = () => {
-  // This will randomly select a killer from the suspects. And add that to the mystery object.
-  mystery.killer = randomSelector(suspects)
+  mystery.killer = randomSelector(suspects);
+  document.getElementById('killerName').innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`;
+  document.getElementById('killerAge').innerHTML = `${mystery.killer.age}`;
+  document.getElementById('killerImage').src = mystery.killer.image
+  document.getElementById('killerOccupation').innerHTML = `${mystery.killer.occupation}`;
+  document.getElementById('killerDescription').innerHTML = `${mystery.killer.description}`;
+}
+//Invoking the function when you click on the killer card:
+document.getElementById('killerCard').addEventListener('click', pickKiller);
 
-  // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
-  document.getElementById('killerCard').style.background = mystery.killer.color
-  document.getElementById(
-    'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
+const pickWeapon = () => {
+  mystery.weapon = randomSelector(weapons);
+  document.getElementById('weaponName').innerHTML = `${mystery.weapon.name}`;
+  document.getElementById('weaponWeight').innerHTML = `${mystery.weapon.weight}kg`;
 }
+document.getElementById('weaponCard').addEventListener('click', pickWeapon);
+
+const pickRoom = () => {
+  mystery.room = randomSelector(rooms);
+  document.getElementById('roomName').innerHTML = `${mystery.room}`;
+}
+document.getElementById('roomCard').addEventListener('click', pickRoom);
 
-// CREATE FUNCTIONS pickWeapon and pickRoom in a similar way.
 
-// STEP 4 - CREATE A FUNCTION revealMystery that will be invoked when you click that button. It should show something like:
-// 'The murder was committed by Jacob Green, in the living room with a rope.'
+//FUNCTION FOR CLICKING ON THE BUTTON THAT REVEAL THE MYSTERY/CRIME
+const revealMystery = () => {
+  document.getElementById('revealButton').innerText = `The murder was committed by ${mystery.killer.firstName} ${mystery.killer.lastName}, in the ${mystery.room} with a ${mystery.weapon.name}.`
```

Comment:
I think it's pretty cool that you changed the button text rather than having another element.  Interesting approach.  
The only drawback is that it's not clear to the user than you can click on the button again, but if you change the cards and press the button again, this function is still called. 👍